### PR TITLE
Add projectile despawning on collision.

### LINKED ISSUE #210: Make Projectiles Despawn on Collision Again
### Description

After #203 is merged we need to make sure projectiles despawn on collisions again, because that functionality got lost in the refactor.

### Alternatives & Prior Art

_No response_

### PR DESCRIPTION
**Pull request by edgarssilva**
Added a new `DespawnOnCollision` component that takes a number of collisions before it is respawned on collision. Added this component to the projectile bundle with the value of 0.

I do not really like the naming on this one, open to ideas :), should fix #210.

### CODE REVIEW

**Review by zicklag — APPROVED** [review]
Looks good!

**Comment by zicklag** [comment]
As far as name, maybe we make it `Breakable`, and then add a field in it for the `hits` or something like that? I've not got any other much better ideas. That one's not easy to name.

**Comment by odecay** [comment]
> As far as name, maybe we make it `Breakable`, and then add a field in it for the `hits` or something like that? I've not got any other much better ideas. That one's not easy to name. man_shrugging

I sort of agree with this but I think if we do that we should consider tying projectile breakability in with item breakability system in #84.

**Comment by edgarssilva** [comment]
> I sort of agree with this but I think if we do that we should consider tying projectile breakability in with item breakability system in #84.

For that we could create a `Drop` component that drops an item when breaking or other conditions like on despawn.

File: src/attack.rs [hunk 1]
```diff
@@ -40,6 +41,19 @@ pub struct Attack {
     pub velocity: Vec2,
 }
 
+/// A component that depawns an entity after collision with tolerance to x collisions.
+#[derive(Component, Clone, Copy, Default, Reflect)]
+pub struct Breakable {
+    pub hits: i32,
+    count: i32,
```
Comment (zicklag) [inline src/attack.rs:48]: My gut reaction is that this should be public. Seems like we might as well allow other systems to modify it it sees fit ( maybe mods for example ). It doesn't really hurt anything other than maybe producing unexpected behavior, which if somebody is modifying that field manually, then it's kind of already obvious what they're messing with.

I you feel like we should keep it private, though, I'm not super set on making it public.

If we do make it public, maybe name the fields, `hit_tolerance` and `hit_count`, and add super minimal doc strings on them to help differentiate what they are for?

**Review by zicklag — APPROVED** [review]
Awesome, looks great!

If you're ready feel free to have bors merge it.

**Pull request merged**